File: Ejercicios/Ejercicio1/1.py
import random

class Auto:
    def __init__(self, nombre):
        self.nombre = nombre
        self.distanciaRecorrida = 0
    
    def avanzar(self):
        distanciaAvance = random.randint(1, 6)  
        self.distanciaRecorrida += distanciaAvance

    def obtenerDistanciaRecorrida(self):
        return self.distanciaRecorrida


class Carrera:
    def __init__(self, distanciaTotal):
        self.distanciaTotal = distanciaTotal
    
    def Carrera(self):
        while jugador.obtenerDistanciaRecorrida() < self.distanciaTotal and computadora.obtenerDistanciaRecorrida() < self.distanciaTotal:
            jugador.avanzar()
            computadora.avanzar()
        
        if jugador.obtenerDistanciaRecorrida() >= self.distanciaTotal and computadora.obtenerDistanciaRecorrida() >= self.distanciaTotal:
            return "Empate "
        elif jugador.obtenerDistanciaRecorrida() >= self.distanciaTotal:
            return "GANASTE CAPO !!"
        else:
            return "PERDISTE =("

jugador = Auto("Tu")
computadora = Auto("PC")
carrera = Carrera(100)
resultado = carrera.Carrera()
print(resultado)

File: Ejercicios/Ejercicio1/1.java
package Ejercicios; 
import java.util.Random;

class Auto {
    private String nombre;
    private int distanciaRecorrida;

    public Auto(String nombre) 
    {
        this.nombre = nombre;
        distanciaRecorrida = 0;
    }

    public void avanzar() {
        Random rd = new Random();
        int distanciaAvance = rd.nextInt(6) + 1;
        distanciaRecorrida += distanciaAvance;
    }

    public int obtenerDistanciaRecorrida() 
    {
        return distanciaRecorrida;
    }
}

class Carrera {
    private int distanciaTotal;

    public Carrera(int distanciaTotal) 
    {
        this.distanciaTotal = distanciaTotal;
    }

    public String empezarCarrera() {
        Auto jugador = new Auto("Tu");
        Auto computadora = new Auto("PC");

        while (jugador.obtenerDistanciaRecorrida() < distanciaTotal && computadora.obtenerDistanciaRecorrida() < distanciaTotal) {
            jugador.avanzar();
            computadora.avanzar();
        }

        if (jugador.obtenerDistanciaRecorrida() >= distanciaTotal && computadora.obtenerDistanciaRecorrida() >= distanciaTotal) {
            return "! EMPATE !";
        } else if (jugador.obtenerDistanciaRecorrida() >= distanciaTotal) {
            return "GANASTE CAPO !!";
        } else {
            return "PERDISTE =(";
        }
    }
}

public class Main 
{
    public static void main(String[] args) {
        Carrera carrera = new Carrera(100);
        String resultado = carrera.empezarCarrera();
        System.out.println(resultado);
    }
}
